src/term/transform/encode_pattern_matching.rs

```diff
@@ -146,40 +146,61 @@ fn make_leaf_pattern_matching_case(
   // The term we're building
   let term = Term::Ref { def_id: rule_def_id };
 
+  // The lambdas  of the rule we're calling
```

Comment:
This comment is not very descriptive.
I think "Whether the args of the rule we're calling are used" is a bit better.